### ISSUE TITLE
Maximum number of JuliaBox instances active error, when no users logged in

### ISSUE THREAD

**Issue by samuelpowell** [JuliaLang/JuliaBox#265]
I get the following error when trying to log into a fresh local installation:
```Maximum number of JuliaBox instances active. Please try after sometime.```

```/engine/log/``` states: ```2015-08-13 11:41:11,121 - ERROR - juliabox.handlers.main.MainHandler - Server maxed out. Can't launch container at hop 11 for user samuelpowell```

```jboxengine/conf/jbox.user``` contains ```"numlocalmax" : 2``` as per installation instructions.

**Comment by tanmaykm** [comment]
Can you do a `docker ps -a` to check how many sessions are active?

Sessions remain active unless you log out of them explicitly from the JuliaBox screen. With `numlocalmax = 2` it is possible to exceed the limit when you are trying with multiple user ids and have forgotten to log out from some of them.

You can configure a session timeout by setting `expire` to a non zero number of seconds of inactivity.

**Comment by samuelpowell** [comment]
I have never been able to log in due to this error, so there are no sessions active. 

```docker ps -a``` returns:
```
CONTAINER ID        IMAGE                        COMMAND                  CREATED             STATUS              PORTS               NAMES
09514e9ff6ae        juliabox/webserver           "/jboxweb/openresty/n"   3 hours ago         Up 3 hours                              webserver_jboxsvc
f1ed1982cc79        juliabox/enginedaemon        "/jboxengine/src/jbox"   3 hours ago         Up 3 hours                              enginedaemon_jboxsvc
e82b1d3639ac        juliabox/engineinteractive   "/jboxengine/src/jbox"   3 hours ago         Up 3 hours                              engineinteractive_jboxsvc
```
followed by a long list of dead sessions. My CPU is currently maxed out running some simulations, could this be a cause?

**Comment by tanmaykm** [comment]
The three listed here are JuliaBox components (names suffixed with jboxsvc), not sessions.

Stopped sessions are also counted because they are likely to be sessions that are getting backed up. A clean logout will not leave behind any container. So yes, that could be the reason.

Maxed out CPU shouldn't be a problem, unless it is causing a visible slowdown Can you stop JuliaBox (`scripts/run/stop.sh`) and then run `scripts/run/docker_rmall.sh` to clean up the dead sessions before starting it up again?

**Comment by samuelpowell** [comment]
I executed ```scripts/run/stop.sh```, then ```scripts/run/docker_rmall.sh```, at which point ```docker ps -a``` shows no containers. I than ran ```scripts/run/start.sh``` and attempted to log in, but I received the same error.

```engineinteractive_err.log``` has more entries for each login attempt, each stating ```2015-08-13 14:58:17,847 - ERROR - juliabox.handlers.main.MainHandler - Server maxed out. Can't launch container at hop 11 for user samuelpowell```

**Comment by samuelpowell** [comment]
```engineinteractive.log``` contains the following entries:

```
2015-08-13 14:58:17,831 - DEBUG - juliabox.jbox_tasks.JBoxAsyncJob - fetching session status from 'localhost'
2015-08-13 14:58:17,831 - DEBUG - juliabox.jbox_tasks.JBoxAsyncJob - sendrecv to tcp://127.0.1.1:8890. connecting...
2015-08-13 14:58:17,843 - DEBUG - juliabox.jbox_tasks.JBoxAsyncJob - sendrecv to tcp://127.0.1.1:8890. received.
2015-08-13 14:58:17,843 - INFO - juliabox.handlers.main.MainHandler - Logged in nowhere
2015-08-13 14:58:17,846 - DEBUG - juliabox.handlers.main.MainHandler - have existing container for samuelpowell_c59e7243e10ce2de188a9643c5a0287228730cd9: False
2015-08-13 14:58:17,846 - DEBUG - juliabox.plugins.compute_singlenode.impl_singlenode.CompSingleNode - Using cached self_stats. Load=100
2015-08-13 14:58:17,847 - DEBUG - juliabox.plugins.compute_singlenode.impl_singlenode.CompSingleNode - Using cached self_stats. Load=100
2015-08-13 14:58:17,847 - DEBUG - juliabox.plugins.compute_singlenode.impl_singlenode.CompSingleNode - Self load: 100
2015-08-13 14:58:17,847 - DEBUG - juliabox.plugins.compute_singlenode.impl_singlenode.CompSingleNode - Not accepting: fully loaded
2015-08-13 14:58:17,847 - DEBUG - juliabox.handlers.main.MainHandler - at hop 11 for user samuelpowell
2015-08-13 14:58:17,847 - ERROR - juliabox.handlers.main.MainHandler - Server maxed out. Can't launch container at hop 11 for user samuelpowell
```
it seems to be upset about the ```self load: 100```, hence my earlier question.

**Comment by tanmaykm** [comment]
Yes You were pointing it out correctly. It wouldn't accept new logins if the machine does not have enough capacity.

**Comment by tanmaykm** [comment]
I think that's a fair behavior to expect. Isn't it?

**Comment by samuelpowell** [comment]
Yes, it is certainly reasonable behaviour. 

Would it be worth making the error message clearer (i.e. 'Maximum load reached', rather than 'Maximum number of instances active')?

My workstations are often running at 100% performing long running (but low priority) simulations, but this is definitely an edge case and I can easily over-ride the default behaviour for my own purposes.

**Comment by tanmaykm** [comment]
Yes. Agree that the error message should be better.

**Comment by samuelpowell** [comment]
Okay, when I have time I'll prepare a PR.
